jspsych.js

```diff
@@ -106,9 +109,20 @@ window.jsPsych = (function() {
         'max_preload_attempts': 10,
         'default_iti': 0,
         'minimum_valid_rt': 0,
-        'experiment_width': null
+        'experiment_width': null,
+        'override_safe_mode': false
       };
 
+      // detect whether page is running in browser as a local file, and if so, disable web audio and video preloading to prevent CORS issues
+      if (window.location.protocol == 'file:' && (options.override_safe_mode === false || typeof options.override_safe_mode == 'undefined')) {
+        options.use_webaudio = false;
+        file_protocol = true;
+        console.warn("jsPsych detected that it is running via the file:// protocol and not on a web server. "+
+          "To prevent issues with cross-origin requests, Web Audio and video preloading have been disabled. "+
+          "If you would like to override this setting, you can set 'override_safe_mode' to 'true' in jsPsych.init. "+
```

Comment:
Maybe the link here could be to even more specific documentation? I wonder if we should have a page in the docs in overview that explains running the file locally vs. a server. We're going to need it for 7.0 anyways, and linking to a more narrative description would probably be clearer.

Reply:
Agree. I was thinking we could add a CORS section to overview/media-preloading, but instead I can add a new page to overview for local vs server that would cover CORS stuff as well as module-related stuff for 7.0. Not sure yet about the page name - let me know if you have suggestions.

Reply:
How about something like How to Run Your Experiment ?

Reply:
We have a new page called "Collecting Data from Online Experiments" that covers hosting options, so would "How to Run Your Experiment" be confusingly similar? What about "Experiment Development" or "Experiment Development Environments"?

Reply:
Right, good point. What if we rename the Collecting Data from Online Experiments to Running Experiments Online and create another page called Running Experiments Offline?

Reply:
I think that naming works. But would it be ok to put them both on the same "Running Experiments" page, with sections for "Offline"(file protocol)  and "Online" (server)? Our current "Online" section is pretty self-contained: options for hosting and recruiting participants. But the "Offline" section that I've drafted is really all about comparisons with "Online": CORS/safe-mode, (lack of) permanent data storage, JavaScript modules, etc.

Reply:
Yeah, that makes total sense.

Reply:
The JS modules thing will wait for the 7.0 docs though, right?

Reply:
Ah yes. I was writing it in now, but it's not relevant yet so including it will just be confusing. I'll save it for the 7.0 docs.

Reply:
@jodeleeuw I just updated this with a draft of the new "Running Experiments" page (replaces the new "Collecting Data from Online Experiments" page - that info is now in the Online section) and I changed the link in the safe mode console warning. I also updated the Media Preloading and Experiment Options pages, since they were out-of-date WRT safe mode and the video stimulus parameter.

Reply:
this new page is awesome! i found one typo, referenced in the comment below. once that is patched let's merge!

Reply:
Yay! Thanks for reviewing!